app/frontend/wallet/shelley/shelley-bitbox02-crypto-provider.ts

```diff
@@ -0,0 +1,280 @@
+import {BitBox02API, getDevicePath, constants} from 'bitbox02-api'
+import * as cbor from 'borc'
+import CachedDeriveXpubFactory from '../helpers/CachedDeriveXpubFactory'
+import {ShelleySignedTransactionStructured, cborizeTxWitnesses} from './shelley-transaction'
+import {hasRequiredVersion} from './helpers/version-check'
+import {BITBOX02_ERRORS, BITBOX02_VERSIONS} from '../constants'
+
+import derivationSchemes from '../helpers/derivation-schemes'
+import {
+  CryptoProvider,
+  CryptoProviderFeature,
+  BIP32Path,
+  HexString,
+  DerivationScheme,
+  AddressToPathMapper,
+  CertificateType,
+} from '../../types'
+import {Network, NetworkId, WalletName} from '../types'
+import {TxSigned, TxAux, CborizedCliWitness} from './types'
+import {InternalError, UnexpectedError, UnexpectedErrorReason} from '../../errors'
+
+type CryptoProviderParams = {
+  network: Network
+  config: any
+}
+
+const ShelleyBitBox02CryptoProvider = async ({
+  network,
+  config,
+}: CryptoProviderParams): Promise<CryptoProvider> => {
+  let bitbox02
```

Comment:
Done. It seems that an existing connection is closed before a new one is opened when unlocking a device.

It would be great if at some point in the future, clicking logout in the UI calls a logout function on the crypto provider, to close the connection then. Otherwise the connection remains open and the device is blocked from being used otherwise, until the website is closed completely. This is not urgent however.